hive-metastore/src/main/java/org/apache/iceberg/hive/MetastoreLock.java

```diff
@@ -0,0 +1,531 @@
+                      lockInfo.lockState = lockFound.lockState;
+                      LOG.info("Found lock {} by agentInfo {}", lockInfo, agentInfo);
+                      return;
+                    }
+                  }
+
+                  throw new LockException(
+                      "Failed to find lock for table %s.%s", databaseName, tableName);
+                } catch (InterruptedException e) {
+                  Thread.currentThread().interrupt();
+                  interrupted.set(true);
+                  LOG.warn(
+                      "Interrupted while trying to find lock for table {}.{}",
+                      databaseName,
+                      tableName,
+                      e);
+                  throw new LockException(
+                      e,
+                      "Interrupted while trying to find lock for table %s.%s",
+                      databaseName,
+                      tableName);
+                }
+              } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                interrupted.set(true);
+                LOG.warn(
+                    "Interrupted while creating lock on table {}.{}", databaseName, tableName, e);
+                throw new LockException(
+                    e, "Interrupted while creating lock on table %s.%s", databaseName, tableName);
+              }
+            },
+            LockException.class);
+
+    // This should be initialized always, or exception should be thrown.
+    LOG.debug("Lock {} created for table {}.{}", lockInfo, databaseName, tableName);
+    return lockInfo;
+  }
+
+  /**
+   * Search for the locks using HMSClient.showLocks identified by the agentInfo. If the lock is
+   * there, then a {@link LockInfo} object is returned. If the lock is not found <code>null</code>
+   * is returned.
+   *
+   * @return The {@link LockInfo} for the found lock, or <code>null</code> if nothing found
+   */
+  private LockInfo findLock() throws LockException, InterruptedException {
+    Preconditions.checkArgument(
+        HiveVersion.min(HiveVersion.HIVE_2),
+        "Minimally Hive 2 HMS client is needed to find the Lock using the showLocks API call");
+    ShowLocksRequest showLocksRequest = new ShowLocksRequest();
+    showLocksRequest.setDbname(databaseName);
+    showLocksRequest.setTablename(tableName);
+    ShowLocksResponse response;
+    try {
+      response = metaClients.run(client -> client.showLocks(showLocksRequest));
+    } catch (TException e) {
+      throw new LockException(e, "Failed to find lock for table %s.%s", databaseName, tableName);
+    }
+    for (ShowLocksResponseElement lock : response.getLocks()) {
+      if (lock.getAgentInfo().equals(agentInfo)) {
+        // We found our lock
+        return new LockInfo(lock.getLockid(), lock.getState());
+      }
+    }
+
+    // Not found anything
+    return null;
+  }
+
+  private void unlock(Optional<Long> lockId) {
```

Comment:
Style:  I see an Intellij warning that Optional is used as an argument.  Optional javadoc states it is only used for return types, is there any way to simplify it?  Maybe lockId can also just be null/non-null?  Reference again:  https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Optional.html  (API Note)

Reply:
I am heavily building on that we either have a `lockId` or not in the method. I can use a parameter which could be `null` and check that, and I can convert the `Optional` from the caller, but it would be awkward. Also since it is a private method, I would keep it his way. WTYT?

Reply:
I see the warnings - actually a few of them:
- 'Optional<Long>' used as type for field 'hmsLockId'
- Result of 'Thread.interrupted()' is ignored
- 'Optional<Long>' used as type for parameter 'lockId'
- Result of 'Thread.interrupted()' is ignored
- Method invocation 'lock' may produce 'NullPointerException'

There was one more which was absolutely correct (null check was missing in `ensureActive`) and I fixed that, but I think the others are fine/intentional.

We can use annotations to turn off them, like `@SuppressWarnings("OptionalUsedAsFieldOrParameterType")`. Does it worth to do it?